invokeai/frontend/src/features/parameters/components/LoraManager/LoraManager.tsx

```diff
@@ -0,0 +1,89 @@
+import { Box } from '@chakra-ui/react';
+import { getLoraModels } from 'app/socketio/actions';
+import { useAppDispatch, useAppSelector } from 'app/storeHooks';
+import IAISimpleMenu, { IAIMenuItem } from 'common/components/IAISimpleMenu';
+import { setPrompt } from 'features/parameters/store/generationSlice';
+import { useEffect, useState, useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+
+export default function LoraManager() {
+  const dispatch = useAppDispatch();
+  const prompt = useAppSelector((state) => state.generation.prompt);
+  const foundLoras = useAppSelector((state) => state.system.foundLoras);
+  const [loraItems, setLoraItems] = useState<IAIMenuItem[]>([
+    { item: '', onClick: undefined },
+  ]);
+  const { t } = useTranslation();
+
+  const loraExists = useCallback(
+    (lora: string) => {
+      const lora_regex = new RegExp(`withLora\\(${lora},?\\s*([\\d.]+)?\\)`);
+      if (prompt.match(lora_regex)) return true;
+      return false;
+    },
+    [prompt]
+  );
+
+  const handleLora = useCallback(
+    (lora: string) => {
+      if (loraExists(lora)) {
+        const lora_regex = new RegExp(`withLora\\(${lora},?\\s*([\\d.]+)?\\)`);
+        const newPrompt = prompt.replace(lora_regex, '');
+        dispatch(setPrompt(newPrompt));
+      } else {
+        dispatch(setPrompt(`${prompt} withLora(${lora},1)`));
+      }
+    },
+    [dispatch, loraExists, prompt]
+  );
+
+  useEffect(() => {
+    dispatch(getLoraModels());
+  }, [dispatch]);
+
+  const renderLoraOption = useCallback(
```

Comment:
Loading the webUI on chrome is causing high CPU load on the InvokeAI tab. Looking at the performance profile, I find this function being called repeatedly many times per second.

Reply:
Looking into it.

Reply:
Pretty sure your issue is from the `useEffect` below on [line 70](https://github.com/invoke-ai/InvokeAI/pull/3072/files/8518f8c2acdaf59ac1948999f0130e725b041548#diff-dae5156119f358589060f8ed64a6ff8ea06f884c4aa8dc1af01f0b5c98b2df84R70). You are listening to `loraItems`, which are being set within the effect. You are making a new array and the effect is checking on strict equality, which means it is never equal.

Changing to `  }, [foundLoras, handleLora, renderLoraOption]);` resolves the issue for me.